packages/kbn-dev-utils/src/precommit_hook/git_utils.ts

```diff
@@ -30,3 +30,29 @@ export async function getGitDir() {
     })
   ).stdout.trim();
 }
+
+// Checks if a correct git version is installed
+export async function isCorrectGitVersionInstalled() {
+  const rawGitVersionStr = (
+    await execa('git', ['--version'], {
+      cwd: REPO_ROOT,
+    })
+  ).stdout.trim();
+
+  const gitVersionMatches = rawGitVersionStr.match(/[0-9]+(\.[0-9]+)+/);
+  const gitVersion =
```

Comment:
If the regex matches then `gitVersionMatches` will be an array containing the capturing groups, otherwise it will be `null`, so all the rematching and length checks seem unnecessary to me.

```ts
const rawGitVersionStr = await execa.stdout('git', ['--version'], {
  cwd: REPO_ROOT,
});

const match = rawGitVersionStr.match(/[0-9]+(\.[0-9]+)+/)
if (!match) {
  return false;
}

const [major, minor] = match[0].split('.').map(n => parseInt(n, 10))
return major > 2 || (major === 2 && minor >= 5)
```

Reply:
You're right, thanks for catching it 🙏